packages/wrangler/pages/functions/template-worker.ts

```diff
@@ -123,7 +161,7 @@ export default {
         // https://fetch.spec.whatwg.org/#null-body-status
         return new Response(
           [101, 204, 205, 304].includes(response.status) ? null : response.body,
-          { ...response, headers: new Headers([...response.headers.entries()]) }
+          { ...response, headers: copyHeaders(response.headers as Headers) }
```

Comment:
I actually think we can simplify this. I must have been tired when I first wrote this, because apparently just doing `new Headers(response.headers)` is enough to correctly make a mutable copy of headers.

As explained to me by the ever-helpful champs: https://discord.com/channels/595317990191398933/910978223968518144/970105175924690994

Reply:
```suggestion
          { ...response, headers: new Headers(response.headers) }
```

Reply:
the new way:

![image](https://user-images.githubusercontent.com/8484333/166443942-9fbf3d55-f3bc-4b03-bcad-56aef86e6780.png)

![image](https://user-images.githubusercontent.com/8484333/166443975-1590ea6f-778d-4a5f-a15b-7da669bbf8ea.png)

vs. the old way:

![image](https://user-images.githubusercontent.com/8484333/166444026-4302342f-888d-457b-bc09-ba47ddf405c3.png)

![image](https://user-images.githubusercontent.com/8484333/166444038-19fa0e08-a1ae-40d6-9e6e-10fa00bf1ad7.png)